### ISSUE TITLE
Returning output tensor without softmax in forward kills the backward pass

### ISSUE THREAD

**Issue by ndrakulicTT** [tenstorrent/tt-forge-fe#554]
In `forge/test/mlir/mnist/training/test_training.py`
Replacing `return nn.functional.softmax(x, dim=-1)` with `return x` with the `batch_size=1`
Chrashes the backward model pass `tt_model.backward()`

```
python: /home/ndrakulic/tt-forge-fe/third_party/tt-mlir/runtime/lib/ttnn/program.cpp:155: void tt::runtime::ttnn::runProgram(::ttnn::MeshDevice &, const ::tt::target::ttnn::Program *, const std::vector< ::ttnn::Tensor *> &, const std::vector< ::ttnn::Tensor *> &): Assertion `inserted && "Duplicate output tensor"' failed.
Fatal Python error: Aborted
```
